### ISSUE TITLE
Crashed Godot Fullscreen Game handling issues

### ISSUE THREAD

**Issue by CitrusWire** [godotengine/godot#44913]
**Godot version:**
3.2.3

**OS/device including version:**
Win 7

**Issue description:**
When a Godot project crashes and it's fullscreen, it exhibits a couple behaviours that seem like bugs.

**Steps to reproduce:**

* Load any project
* Project settings > full screen
* Introducing a parsing error into it
* Execute

**What does Happen**

1) Godot fills the screen with black (or with other projects (https://github.com/GDQuest/godot-2d-tactical-space-combat) it does not, and runs invisibly in the background).

2) Alt-tab out of the black screen, then when switch back to the running game, the entire screen locks up.

**What Should happen**
* In relation to (1), if the application has a parse error, it shouldn't even try and load to that level. There's no point because it has already crashed.

* In relation to (2), this gives the impression that the entire computer has crash. When you switch back to the game the game takes up rendering of the screen, but is transparent, so you see exactly what you saw before switching back into it. Except that most inputs are ignored so it feels like a lock-up. Alt-tab and other system-wide shortcuts still work fortunately.
If the application has crashed, I'm not sure what it should do, but it shouldn't do this.

**Minimal reproduction project:**
n/a

**Comment by Calinou** [comment]
> Alt-tab out of the black screen, then when switch back to the running game, the entire screen locks up.

This is expected, as the game is no longer drawing when it's paused.

**Comment by CitrusWire** [comment]
> > Alt-tab out of the black screen, then when switch back to the running game, the entire screen locks up.
> 
> This is expected, as the game is no longer drawing when it's paused by the editor.

Which makes sense, but from the user's perspective, it's really confusing and seems like a bug. The first time it happened I thought Godot had crashed. Then I thought it was just locking up. Only after I'd half written a bug report did I realise it was a consequence of the game being paused.

I'd suggest in this scenario if it's unrecoverable it should either exit full screen, or simply terminate entirely. If there are plausible debugging scenarios where that's undesirable, maybe have an editor setting to override (or the opposite, have a new setting for "close app on crash" and enable it by default).

(In my case it was compounded by the fact that the first time I saw this behaviour was by importing a 3rd party project, so I had no reason to expect a crash)

**Comment by Calinou** [comment]
> or simply terminate entirely.

If we do that, you won't be able to debug the paused game process anymore.

**Comment by CitrusWire** [comment]
> If we do that, you won't be able to debug the paused game process anymore.

Which makes me think of something else; strictly speaking it's an enhancement, but directly relevant here:

When Godot Editor freezes the game during debugging (let alone a crash like this), the application focus should switch back to the Editor. After all, it's impossible to interact with the game until the editor resumes. It does when it hits the first breakpoint, but not an subsequent ones.

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot/issues/80798, which is the actual bug.
